### PR TITLE
chore: add generic methods and use them part 2/2

### PR DESCRIPTION
**Pull request by DmitriyMV**
Use things from #5702.

Signed-off-by: Dmitriy Matrenichev <dmitry.matrenichev@siderolabs.com>

---
This change is [Reviewable](https://reviewable.io/reviews/siderolabs/talos/5720)

### CODE REVIEW

File: internal/pkg/configuration/configuration.go [hunk 1]
```diff
@@ -53,7 +53,8 @@ func Generate(ctx context.Context, in *machine.GenerateConfigurationRequest) (re
 						DeviceCIDR:      device.Cidr,
 						DeviceDHCP:      device.Dhcp,
 						DeviceIgnore:    device.Ignore,
-						DeviceRoutes:    routes,
+						// TODO: this looks strange, since its always empty de-facto.
```
Comment (DmitriyMV) [inline internal/pkg/configuration/configuration.go:56]: I have no idea whats happening here

Comment (smira) [reply internal/pkg/configuration/configuration.go:56]: @Unix4ever should know! I think it should be a copy, but ?

**Review by smira — APPROVED** [review]
:sunglasses:

**Comment by DmitriyMV** [comment]
I added 

```go
DeviceRoutes: slices.Map(device.Routes, func(route *machine.RouteConfig) *v1alpha1.Route {
	return &v1alpha1.Route{
		RouteNetwork: route.Network,
		RouteGateway: route.Gateway,
		RouteMetric:  route.Metric,
	}
}),
```

**Comment by DmitriyMV** [comment]
FYI @Unix4ever

**Pull request merged**